src/nvim/api/buffer.c

```diff
@@ -1134,9 +1134,63 @@ Boolean nvim_buf_is_valid(Buffer buffer)
   return ret;
 }
 
+/// Deletes a named mark in the buffer
+//
+/// Marks are (1,0)-indexed. |api-indexing|
+///
+/// @note Only deletes marks associated with the buffer
+///
+/// @param buffer     Buffer to set the mark on
+/// @param name       Mark name {a-zZ0-9}
+void nvim_buf_del_mark(Buffer buffer, String name, Error *err)
+  FUNC_API_SINCE(8)
+{
+  buf_T *buf = find_buffer_by_handle(buffer, err);
+  if (!buf) {
+    return;
+  }
+  if (name.size != 1) {
+    api_set_error(err, kErrorTypeValidation,
+                  "Mark name must be a single character");
+  }
+  // pos->lnum is 0 when the mark is not valid in the buffer, or is not set.
+  pos_T *pos = getmark_buf(buf, *name.data, false);
+  if (pos->lnum != 0) {
+    // since the mark belongs to the buffer delete it.
+    set_mark(buf, name, 0, 0, err);
+  }
+}
+
+/// Sets a named mark in the given buffer
```

Comment:
why only named? we should at least leave the door open for other types in the future.

Reply:
Since there's already `nvim_buf_set_extmark` which is the other kind of mark, and we also have different indexing schemes, seems confusing. In the other hand in the future there might a need for custom named marks (like bookmarks maybe?), so that might be an option.

Reply:
I'm thinking of special marks like `'` , not extmarks.

> In the other hand in the future there might a need for custom named marks (like bookmarks maybe?),

Yeah. Might want to put a `Dictionary opt` parameter on all/most of these functions, for future expansion. (See other functions for examples of this pattern)

Reply:
This functions cover all types of marks. Since it's using `setmark_pos` this changes all types. 

https://github.com/neovim/neovim/blob/0aa8128aaa9e6f714372f7ea7299f4115de0a962/src/nvim/mark.c#L101

 I'm under the impression that all default vim marks are called "named marks"  I'll check in the docs to make sure.